Unified Talent Management, AutoGear & Wipe Enhancements

Wishmaster117 opened the pull request:
This PR introduces a comprehensive upgrade to the MultiBot system by combining three key features into a single, cohesive update:

🧠 Intelligent Talent Spec Management

🛠️ Accurate Active Talent Tree Rendering

🎒 AutoGear Button & 🔄 Wipe Function on the Every Bar

🧠 Intelligent Talent Spec Management
This feature allows players to manage each bot’s talent specialization directly from the UI with smart automation and intuitive controls.

🔧 Features:
Auto-detection of available specs for each bot via /w talents spec list.

Current active spec is identified via /w talents, with visual indication (grayed-out button).

One button per available spec with:

Class icon and tooltip (PvE/PvP purpose).

Left-click = Install spec in slot 1 (primary).

Right-click = Install spec in slot 2 (secondary).

Bot casting/eating is interrupted automatically before spec switching.

New “Set Talents” button added to each bot frame.

🔄 Talent UI Now Reflects Actual Active Spec
Previously, the UI always displayed spec 1 even when spec 2 was active. This has been fixed with dynamic detection and rendering.

✅ Improvements:
Dynamic spec rendering based on bot’s actual active spec.

Live updates when bots switch spec in-game.

Refactored rendering logic via a single RenderActiveTalentTree() method.

Optimized performance:

Batched UI redraws.

Removed outdated and unused logic.

Thoroughly tested across multiple classes (Mage, Priest, Warrior).

🧰 Every Bar: AutoGear & Wipe Buttons
Enhances the "Every" action bar by adding direct control tools:

🎒 AutoGear Button
Each bot now has an AutoGear button.

Triggers smart gear installation based on predefined thresholds (quality, GS).

Fully localized for all supported languages.

☠️ Wipe Button
New Wipe button forcibly kills and resurrects a selected bot.

Resets bot’s position, health, and mana without needing combat or downtime.

Useful for unsticking, resetting or re-syncing bots in one click.

🌍 Other Notes
Full localization across all available MultiBot languages (enUS, frFR, deDE, koKR, ruRU, zhCN, esES, etc.).

No ACE3 dependency.

Fully backward-compatible and non-intrusive to existing behaviors.

Thank you for reviewing!
This unified PR brings much-requested QoL improvements, performance optimizations, and deeper bot control to the MultiBot experience.
Please let me know if you'd like any part split, adjusted, or further documented.

<img width="1540" height="1156" alt="image" src="https://github.com/user-attachments/assets/664fc087-5081-464f-a7c6-d288e8f90a87" />

<img width="1454" height="1203" alt="image" src="https://github.com/user-attachments/assets/57f7aa95-d95b-45ba-beeb-ff114d8f3607" />

<img width="1555" height="686" alt="image" src="https://github.com/user-attachments/assets/5441818d-0332-4378-a4b0-5381ee4b09be" />

<img width="1358" height="769" alt="image" src="https://github.com/user-attachments/assets/2000a650-1e92-4a9d-aa9a-307a2fefa8c9" />

Wishmaster117 committed:
More functions added

Wishmaster117 committed:
Update MultiBotSpecUI.lua

brighton-chi commented:
Thank you. A couple of thoughts.

(1) If there will be an autogear button, can it be optional to show? That would be a bad action to take by accident so I'd like to avoid that risk. I don't know if it's really needed in the first place since I don't think autogear would be a very frequently used command by most people, but that's neither here nor there for me.

(2) It could be helpful to include buttons relating to the recent updates to warlock strategies (namely, setting curse, pet, and soulstone strategies).

Wishmaster117 committed:
Add a Misc button in Bot's Every Bar with sub-buttons

Wishmaster117 committed:
Update MultiBot.lua

Wishmaster117 commented:
> Thank you. A couple of thoughts.
> 
> (1) If there will be an autogear button, can it be optional to show? That would be a bad action to take by accident so I'd like to avoid that risk. I don't know if it's really needed in the first place since I don't think autogear would be a very frequently used command by most people, but that's neither here nor there for me.
> 
> (2) It could be helpful to include buttons relating to the recent updates to warlock strategies (namely, setting curse, pet, and soulstone strategies).

Hi,

I do that:

Grouped “Autogear” & “Wipe” under a new “Misc” dropdown

Added a parent “Misc” button that toggles a small submenu above it.

Moved the existing Autogear and Wipe buttons into this submenu, preserving their original callbacks. (no miss click possible)

Made the “SetTalents” button a true toggle

Removed the right‐click handler.

On left‐click: if the talent dropdown is closed, it is requested and the button is enabled; if open, it is hidden and the button is disabled.

Initialized the button in its disabled state.

Updated tooltips for clarity and localization

Improved English tooltip copy for SetTalents, AutoGear, and Misc.

Provided fully localized versions of each tooltip in German, Spanish, French, Russian, Korean, and Simplified Chinese.

These changes keep existing behavior intact while improving UI discoverability and internationalization support. Let me know if you’d like any further tweaks!

Wishmaster117 commented:
I already finished the custom talents apply interface,  glyphs interface and apply custom glyphs interface .
Just waiting if my PR in PlayerBots github are accepted: https://github.com/liyunfan1223/mod-playerbots/pull/1461#issue-3253142166
 and i post the PR.

Wishmaster117 commented:
> (2) It could be helpful to include buttons relating to the recent updates to warlock strategies (namely, setting curse, pet, and soulstone strategies).

Done and for Hunters Too
